### ISSUE TITLE
Select now() returns wrong time

### ISSUE THREAD

**Issue by greyscaled** [GlareDB/glaredb#1427]
## Context

```
greyb ~ $ ./glaredb local
GlareDB (v0.2.1)
Type \help for help.
Using in-memory catalog
〉select now();
```

## Expected

```console
postgres=> select now();
              now
-------------------------------
 2023-07-29 16:32:31.782682+00
(1 row)
```

## Actual

```console
+----------------------+
| now()                |
+----------------------+
| 1970-01-01T00:00:00Z |
+----------------------+
```

**Comment by greyscaled** [comment]
Furthermore:

```sql
select now();
```

On cloud causes disconnect from the compute engine

**Comment by greyscaled** [comment]
I'm going to move this out of `v0.3.0` since we fixed the panic here: #1431

**Comment by scsmithr** [comment]
Unassigning myself, as it will be solved by #1432 and I haven't committed to taking that on.
